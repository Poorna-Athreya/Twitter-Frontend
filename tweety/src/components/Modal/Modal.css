.modal{
    margin: 2%;
    background-color: #DFDFDF;
    width: 50%;
    padding-bottom: 2%;
}
.cross-btn{
    background-color: var(--card-bg-dark);
    float: right;
    border: none;
    width: 10%;
    height: 30px;
    color: white;
}
.cross-btn:hover{
    background-color: red;
}
.new-tweet-box{
    padding-top: 2%;
    text-align: center;
}
.new-tweet-text-box{
    width: 80%;
    height: 100px;
}
.post-new-tweet{
    width: 30%;
    height: 50px;
    border: none;
    font-weight: bold;
    font-size: 1.2em;
    color:white;
    background-color: #0080FF;
    margin-left: 9.5%;
}
@media screen and  (max-width:400px){
    .modal{
        width: 80%;
        margin: 0px;
        margin-top: 1%;
    }
}