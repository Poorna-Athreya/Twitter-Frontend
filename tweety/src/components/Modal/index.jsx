import React, { useState } from 'react';
import PropTypes from 'prop-types';
import './Modal.css';

export default function Modal({ setShowModal, modalSubmitHandler }) {
  const [newTweet, setNewTweet] = useState('');

  const onChangeHandler = (event) => {
    setNewTweet(event.target.value);
  };

  const onPost = (event) => {
    event.preventDefault();
    modalSubmitHandler(newTweet);
    setNewTweet('');
  };
  return (
    <div className="modal">
      <button data-testid="modalCloseButton" type="button" onClick={() => setShowModal(false)} className="cross-btn">X</button>
      <div className="new-tweet-box">
        <h3 data-testid="modalHeading">Add New Tweet</h3>
        <br />
        <input data-testid="modalInput" type="text" value={newTweet} className="new-tweet-text-box" onChange={onChangeHandler} />
      </div>
      <br />
      <button data-testid="modalSubmitButton" type="submit" className="post-new-tweet" onClick={onPost}>Post</button>
    </div>
  );
}
Modal.propTypes = {
  setShowModal: PropTypes.func.isRequired,
  modalSubmitHandler: PropTypes.func.isRequired,
};
