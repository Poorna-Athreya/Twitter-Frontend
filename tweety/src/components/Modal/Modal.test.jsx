/* eslint-disable max-len */
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import Modal from './index';

const mockSetShowModal = jest.fn();
const mockModalSubmitHandler = jest.fn();
const mockNavigate = jest.fn();
const mockParams = jest.fn().mockResolvedValue(2);
jest.mock('react-router-dom', () => ({ useNavigate: () => mockNavigate, useParams: () => mockParams }));

describe('Modal', () => {
  beforeEach(() => {
    render(<Modal setShowModal={mockSetShowModal} modalSubmitHandler={mockModalSubmitHandler} />);
    mockSetShowModal.mockClear();
    mockNavigate.mockClear();
    mockParams.mockClear();
  });

  it('should take a snapshot for Modal', () => {
    const { asFragment } = render(<Modal setShowModal={mockSetShowModal} modalSubmitHandler={mockModalSubmitHandler} />);
    expect(asFragment(<Modal setShowModal={mockSetShowModal} modalSubmitHandler={mockModalSubmitHandler} />)).toMatchSnapshot();
  });
  it('should have the required elements when Modal is rendered with the setShowModal Function', () => {
    expect(screen.queryByTestId('modalCloseButton')).toBeTruthy();
    expect(screen.queryByTestId('modalHeading')).toBeTruthy();
    expect(screen.queryByTestId('modalInput')).toBeTruthy();
    expect(screen.queryByTestId('modalSubmitButton')).toBeTruthy();
  });
  it('should change the modal input value when input text changes', () => {
    const mockModalInput = 'new tweet';
    fireEvent.change(screen.getByTestId('modalInput'), {
      target: { value: mockModalInput },
    });
    expect(screen.getByTestId('modalInput')).toHaveAttribute(
      'value',
      mockModalInput,
    );
  });
  it('should call the mock modal submit handler function with the mock input value when the submit button is clicked', () => {
    const mockModalInput = 'new tweet';
    fireEvent.change(screen.getByTestId('modalInput'), {
      target: { value: mockModalInput },
    });
    expect(screen.getByTestId('modalInput')).toHaveAttribute(
      'value',
      mockModalInput,
    );
    fireEvent.click(screen.getByTestId('modalSubmitButton'));
    expect(mockModalSubmitHandler).toHaveBeenCalledTimes(1);
    expect(mockModalSubmitHandler).toHaveBeenCalledWith(mockModalInput);
  });
  it('should call the mockSetShowModal with value false when the close button is clicked', () => {
    fireEvent.click(screen.getByTestId('modalCloseButton'));
    expect(mockSetShowModal).toHaveBeenCalledTimes(1);
    expect(mockSetShowModal).toHaveBeenCalledWith(false);
  });
});
