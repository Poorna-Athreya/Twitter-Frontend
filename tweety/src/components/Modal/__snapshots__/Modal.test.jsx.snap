// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should take a snapshot for Modal 1`] = `
<DocumentFragment>
  <div
    class="modal"
  >
    <button
      class="cross-btn"
      data-testid="modalCloseButton"
      type="button"
    >
      X
    </button>
    <div
      class="new-tweet-box"
    >
      <h3
        data-testid="modalHeading"
      >
        Add New Tweet
      </h3>
      <br />
      <input
        class="new-tweet-text-box"
        data-testid="modalInput"
        type="text"
        value=""
      />
    </div>
    <br />
    <button
      class="post-new-tweet"
      data-testid="modalSubmitButton"
      type="submit"
    >
      Post
    </button>
  </div>
</DocumentFragment>
`;
